### ISSUE TITLE
 one atom appears twice in list

### ISSUE THREAD

**Issue by VincenzoChen** [getcontacts/getcontacts#68]
Hello,sir! I met such error when I ran the program. I am confused that how this error come into being and how can I fix it?

**Comment by RasmusFonseca** [comment]
Hi Vincenzo. Could you give an example of the output you're seeing and if possible the commands that led to it?

**Comment by VincenzoChen** [comment]
Actually I have met two problems.   
The first one is when I specify ligand as 23Q in command like this`get_dynamic_contacts.py --topology 3kwj.pdb --trajectory 3kwj.dcd --output 3kwj_interaction.tsv --ligand 23Q --sele "resid 39 to 766" --itypes sb pc ps ts hb vdw`, the output is:
![error1](https://user-images.githubusercontent.com/40376903/82107527-164a2f00-975b-11ea-8ea7-8c878fa26af0.png)
It says that it is unable to create macro for 'ligand'.    
The second one is stated as this issue. I did not specify the ligand.Then the problem is like this:
![error2](https://user-images.githubusercontent.com/40376903/82107568-69bc7d00-975b-11ea-849b-d65308f1345f.png)
When I reduce the distance parameter in function extract_ligand_features of contacts_utils.py, it can run smoothly.
![ND~F PU4`4PX%$_TUH7P72R](https://user-images.githubusercontent.com/40376903/82107634-efd8c380-975b-11ea-9686-04f4e44dd540.png)

**Comment by VincenzoChen** [comment]
The second problem i have already solved.But the first i still don't know how to fix it.

**Comment by T00mek** [comment]
Dear Rasmus and Vincenzo I am running into the same problem. In my case it almost always applies to the last atom of the ligand, the error code looks like this:
![image](https://user-images.githubusercontent.com/45410138/83059851-d89ead80-a05a-11ea-9074-a66682987a6f.png)

However in the topology ffile there is no indicaiton that this is adouble:

if i delet atom CL2 it runs smoothly:

![image](https://user-images.githubusercontent.com/45410138/83059948-fe2bb700-a05a-11ea-8807-61426d264e8d.png)

**Comment by RasmusFonseca** [comment]
Thanks for the reports. I'll check over the weekend if I can reproduce

**Comment by T00mek** [comment]
Ok, if its necessary I can send over the trajecotyr/topology

**Comment by VincenzoChen** [comment]
![bug](https://user-images.githubusercontent.com/40376903/83091919-dd04ae00-a0ce-11ea-9a4f-1aa3d95c0388.png)
I think the second for cyclic sentence is not necessary. So I just deleted it and the error dissappeared.

**Comment by VincenzoChen** [comment]
![bug](https://user-images.githubusercontent.com/40376903/83092330-a713f980-a0cf-11ea-83b2-eb971e303b83.png)
I think the second for cyclic sentence is not necessary. So I just deleted it and the error dissappeared

**Comment by RasmusFonseca** [comment]
@VincenzoChen did you fix the first issue with the macro not defined?

**Comment by RasmusFonseca** [comment]
@VincenzoChen I couldn't reproduce the second problem. Very happy to check again if you could send me the input files.

**Comment by RasmusFonseca** [comment]
@T00mek it'd be great if you could send top/trj file that reproduces this problem. I can't reproduce right now.

**Comment by VincenzoChen** [comment]
> @VincenzoChen did you fix the first issue with the macro not defined? I think the problem is that the 23Q is not a valid VMD selection. Try putting `--ligand "resname 23Q"`

I fix this issue by rewrite the sentence to select ligand in function configure_ligand. Here I show in the picture which is highlighted by red line.
![bug1](https://user-images.githubusercontent.com/40376903/83380863-a6ea6580-a411-11ea-8e63-7633c13b924e.png)

**Comment by VincenzoChen** [comment]
> @VincenzoChen I couldn't reproduce the second problem. Very happy to check again if you could send me the input files.

For the second problem, I solved it by deleting a for cyclic sentence in extract_ligand_featuresis function because I think it is not necessary. Here I show it in the picture which is highlighted by red line.
![bug](https://user-images.githubusercontent.com/40376903/83381084-3ee84f00-a412-11ea-864d-cf7b38fa85a6.png)

**Comment by RasmusFonseca** [comment]
@VincenzoChen I agree that line shouldn't be necessary, but I also don't see how removing it changes the result of the function. @agchempa would be great if you could take a look.
